File: api/src/main/java/com/ruby/rubia_server/core/dto/UpdateCompanyDTO.java
package com.ruby.rubia_server.core.dto;

import com.ruby.rubia_server.core.enums.CompanyPlanType;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UpdateCompanyDTO {

    @Size(min = 2, max = 255, message = "Nome deve ter entre 2 e 255 caracteres")
    private String name;

    @Size(max = 1000, message = "Descrição não pode exceder 1000 caracteres")
    private String description;

    @Email(message = "Email de contato deve ter formato válido")
    private String contactEmail;

    @Pattern(regexp = "^\\+?[0-9.()\\-\\s]*$", message = "Telefone de contato inválido")
    private String contactPhone;

    private String logoUrl;

    private Boolean isActive;

    private CompanyPlanType planType;

    private Integer maxUsers;

    private Integer maxWhatsappNumbers;
}

File: api-dotnet/Rubia.Server/DTOs/UpdateCompanyDto.cs
using System.ComponentModel.DataAnnotations;
using Rubia.Server.Enums;

namespace Rubia.Server.DTOs;

public class UpdateCompanyDto
{
    [StringLength(255, MinimumLength = 2, ErrorMessage = "Nome deve ter entre 2 e 255 caracteres")]
    public string? Name { get; set; }

    [StringLength(1000, ErrorMessage = "Descrição não pode exceder 1000 caracteres")]
    public string? Description { get; set; }

    [EmailAddress(ErrorMessage = "Email de contato deve ter formato válido")]
    public string? ContactEmail { get; set; }

    public string? ContactPhone { get; set; }

    public string? LogoUrl { get; set; }

    public bool? IsActive { get; set; }

    public CompanyPlanType? PlanType { get; set; }

    public int? MaxUsers { get; set; }

    public int? MaxWhatsappNumbers { get; set; }

    public int? MaxAiAgents { get; set; }
}